### PR TITLE
Templatized ModuleWithProviders in typescript-angular

### PR DESCRIPTION
**Pull request by bezineb5**
The call ModuleWithProviders requires a template: https://angular.io/api/core/ModuleWithProviders
Apparently, in Angular 10, this is strictly enforced

### CODE REVIEW

File: src/main/resources/handlebars/typescript-angular/api.module.mustache [hunk 1]
```diff
@@ -18,7 +18,7 @@ import { {{classname}} } from './{{importPath}}';
     {{/hasMore}}{{/apis}}{{/apiInfo}} ]
 })
 export class ApiModule {
-    public static forRoot(configurationFactory: () => Configuration): ModuleWithProviders {
+    public static forRoot(configurationFactory: () => Configuration): ModuleWithProviders<ApiModule> {
```
Comment (jose-hernandez-iop) [inline src/main/resources/handlebars/typescript-angular/api.module.mustache:21]: I think that for consistency with V2 and to allow backwards compatibility with versions of Angular <10 this should be:

`public static forRoot(configurationFactory: () => Configuration): ModuleWithProviders{{#genericModuleWithProviders}}<ApiModule>{{/genericModuleWithProviders}} {`

Comment (bezineb5) [reply src/main/resources/handlebars/typescript-angular/api.module.mustache]: Thanks for the feedback!
I actually copied the method from https://github.com/swagger-api/swagger-codegen/pull/10464 which automatically sets genericModuleWithProviders if angular is >= 7.

**Comment by jose-hernandez-iop** [comment]
@HugoMario any chance this PR could be merged in?

**Comment by raman-nbg** [comment]
Why is this not merged? With Angular 12 it is not possible to use `ModuleWithProviders` without generic type parameter.

**Comment by lab900-simon-dethuin** [comment]
Can we get this merged?

**Comment by doraemoncito** [comment]
It would be good to see this change merged in.  The swagger generator for Angular is really falling behind the current version of Angular.

**Comment by KillDozerX2** [comment]
Why is this PR still not merged?

**Comment by hbtjma** [comment]
Who is responsible for typescript-angular and is this person still active?

**Comment by HugoMario** [comment]
Thanks for PR, and sorry for this big delay :/

**Pull request merged**